source/MRMesh/MRMeshLoadObj.cpp

```diff
@@ -747,9 +756,14 @@ Expected<std::vector<NamedMesh>, std::string> fromSceneObjFile( const char* data
         }
         std::sort( ordTrisAll.begin(), ordTrisAll.end(),
                    [] ( auto&& a, auto&& b ) { return a.offset < b.offset; } );
-        t.reserve( t.size() + trisSize );
+
+        triangulation.reserve( triangulation.size() + trisSize );
+        texturePerFace.reserve( triangulation.size() + trisSize );
```

Comment:
It's better not to allocate excess memory in case no materials/textures are associated with the file.